contracts/SnappBase.sol

```diff
@@ -117,6 +147,72 @@ contract SnappBase is Ownable {
 
         stateRoots.push(_newStateRoot);        
         depositHashes[slot].applied = true;
-        emit StateTransition(TransitionType.Deposit, _currStateRoot, _newStateRoot, slot);
+        emit StateTransition(TransitionType.Deposit, this.stateIndex(), _newStateRoot, slot);
+    }
+
+    function requestWithdrawal(uint8 tokenId, uint amount) public onlyRegistered() {
+        require(amount != 0, "Must request positive amount");
+
+        address tokenAddress = tokenIdToAddressMap[tokenId];
+        require(tokenAddress != address(0), "Requested token is not registered");
+        require(
+            ERC20(tokenAddress).balanceOf(address(this)) >= amount,
```

Comment:
I don't know whether these kinds of sanity checks make sense. In most cases, they will just cost gas.

Reply:
But we would somehow need to check that the amount is not generating an overflow. In the snark, we will only be able to deal with a certain precision.

Reply:
What do you mean by overflow here? There is a limit to the number of withdraws per slot?

Reply:
no..., the variable amount will be represented by some amount of bits, maybe 100, but the uint256 has 256 bits. So there is the possibility to overflow the snark 100 bits

Reply:
Do we have any estimate on what the snark can handle?